### ISSUE TITLE
Consistent get/setParameters

### ISSUE THREAD

**Issue by fredludlow** [arose/ngl#239]
I've been trying to write code to copy parameters:
```
var p = repr1.getParameters();
repr2.setParameters(p);
```
This fails for some representation types as some of the parameter definitions are nulled out, e.g.
```
    parameters: Object.assign( { 
    ...
    },  Representation.prototype.parameters, {

        flatShaded: null,
        side: null,
        wireframe: null,
    })
```

There's a check when actually setting the parameters, but it checks for undefined not null:
https://github.com/arose/ngl/blob/master/src/representation/representation.js#L458

(The actual error is line 460 where `undefined` has no property `int`):
`           if( tp[ name ].int ) p[ name ] = parseInt( p[ name ] );`

Possible fixes I can think of:
1) It's a silly idea, don't do it. 
2) Use `undefined` instead of `null` in sub-classes:
```
    Object.assign( { },  Representation.prototype.parameters, {

        flatShaded: undefined,
        side: undefined,
        wireframe: undefined,
```
3) Don't report undefined/null parameters in `getParameters`
```
getParameters: function(){

        var params = {
            lazy: this.lazy,
            visible: this.visible,
            quality: this.quality
        };

        Object.keys( this.parameters ).forEach( function( name ){
            if( this.parameters[ name ] === null ){ return; } // Add this line?
            if( this.parameters.type === "button" ){
                params[ name ] = this[ name ].bind( this );
            }else{
                params[ name ] = this[ name ];
            }
        }, this );

        return params;
```

I've gone with 3 as a temporary fix here, what do you reckon?

**Comment by arose** [comment]
Nice find, thanks! Fixed here 287abe9db5225a8e5a62ea219b6bbecf8a46df57

**Comment by fredludlow** [comment]
Great, thanks!
